vspec.py

```diff
@@ -225,15 +244,31 @@ def yaml_compose_node(parent, index):
         except yaml.parser.ParserError as e:
             raise VSpecError(file_name, line + 1, e)
 
-        node.__line__ = line + 1
-        node.__file_name__ = file_name
+        if node.value == '$include$':
+            node.value = f'$include${load_flat_model.include_index}'
+            load_flat_model.include_index = load_flat_model.include_index + 1
+
+        # Avoid having root-level line numbers as non-dictionary entries
+        if parent:
+            node.__line__ = line + 1
+            node.__file_name__ = file_name
+        else:
+            node.__line__ = None
+            node.__file_name = None
         return node
 
-    def yaml_construct_mapping(node, deep=False):
+
+
+    def yaml_construct_mapping(node, deep=True):
         mapping = yaml.constructor.Constructor.construct_mapping(loader, node, deep=deep)
-        # Find the name of the branch / signal and convert
-        # it to a dictionary element '$name$'
+
+        # Replace
+        # { 'Vehicle.Speed': { 'datatype': 'boolean', 'type': 'sensor' }}
+        # with
+        # { '$name$': 'Vehicle.Speed', 'datatype': 'boolean', 'type': 'sensor' }
+
         for key, val in list(mapping.items()):
+            # Don't convert metadata.
```

Comment:
Moderately confusing comment since we often call the listed attributes in VSS (datatype, description, ...) "metadata".  Here you mean maybe something like 'special directives' ?